narwhal/worker/src/quorum_waiter.rs

```diff
@@ -118,10 +118,13 @@ impl QuorumWaiter {
                                 total_stake += stake;
                                 if total_stake >= threshold {
                                     let digest = batch.digest();
+                                    let timestamp = batch.metadata.timestamp;
```

Comment:
small nit/feel free to ignore because it's just my personal style preference: `timestamp` could probably be inlined in the struct below? It looks like `digest` isn't inlined becuase it's used twice, and we probably don't want to pay to compute it twice, but timestamp only appears once and adding an extra name for it here doesn't seem to help clarity?

Reply:
Yep you are right , I did try to follow a more uniform approach in extracting the necessary info - but as you said not much of a point probably.